ibis/expr/datatypes/value.py

```diff
@@ -233,8 +233,9 @@ def infer_shapely_multipolygon(value) -> dt.MultiPolygon:
 
 
 @public
-class _WellKnownText(NamedTuple):
```

Comment:
When this was a `NamedTuple` the pattern matching traversal would recurse into it, resulting in issues reconstructing geospatial literal nodes.

I tried fixing a few places where it was checking for `isinstance(val, tuple)` to `type(val) is tuple` in the pattern handling code, but it endded up being easier to have this type not be a tuple type. I still think we might want to change those traversals to skip namedtuples (and maybe all subclasses of builtin containers).

Reply:
Could make sense, performance-wise too.

Reply:
Should I open an issue about avoiding recursing into non-builtin types on the patterns/rewrites?